### PR TITLE
feat(cli): add CLI command to serve a runner

### PR DESCRIPTION
**Pull request by bojiang**
to arrange our serving CLI/Python APIs like this:
```
bentoml serve
bentoml serve --production
bentoml serve-grpc
bentoml serve-grpc --production

bentoml start-grpc-server
bentoml start-rest-server
bentoml start-runner-server
```
The APIs start with start are low-level APIs used in Yatai, would be hidden from the help message for now 

run bare servers from container
`docker run -it --rm -p 3000:3000 -e BENTOML_SERVE_COMPONENT=rest_server -e BENTOML_SERVE_RUNNER_MAP="asdfasdf=1 py_model.case-1.e2e=tcp://192.168.67.1:4000" general_workflow_service.case-1.e2e:tefgtebeok7iiusu`
`docker run -it --rm -p 4000:3000 -e BENTOML_SERVE_COMPONENT=runner -e BENTOML_SERVE_RUNNER_NAME=py_model.case-1.e2e general_workflow_service.case-1.e2e:awvtyqreokd4cusu`

### CODE REVIEW

File: bentoml_cli/worker/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Here are entrypoints for the bare workers. They are typically used by the supervisor and not directly by the user.
+
```
Comment (aarnphm) [inline bentoml_cli/worker/README.md:2]: To run a standlone api server do
```bash
python -m bentoml_cli.worker.http_api_server ...
```

File: bentoml/_internal/bento/docker/entrypoint.sh [hunk 2]
```diff
@@ -11,7 +11,7 @@ _is_sourced() {
 
 _main() {
 	# if first arg looks like a flag
-	if [ "${1:0:1}" = '-' ]; then
+	if [ -z "$@" ] || [ "${1:0:1}" = '-' ]; then
```
Comment (ssheng) [inline bentoml/_internal/bento/docker/entrypoint.sh:14]: Is `-z "$@"` a superset of `"${1:0:1}" = '-'`?

File: bentoml/serve.py [hunk 3]
```diff
@@ -393,3 +393,202 @@ def serve_production(
...
+                port,
+            ),
+        )
+    finally:
+        if uds_path is not None:
+            shutil.rmtree(uds_path)
+
+
+@inject
+def serve_bare_api_server(
```
Comment (ssheng) [inline bentoml/serve.py:485]: "Serve" isn't technically accurate here. How about just `start_api_server` and `start_runner`?

File: bentoml_cli/serve.py [hunk 4]
```diff
@@ -17,6 +17,30 @@ def add_serve_command(cli: click.Group) -> None:
 
     @cli.command()
     @click.argument("bento", type=click.STRING, default=".")
+    @click.option(
+        "--component",
+        type=click.STRING,
+        default=None,
+        envvar="BENTOML_SERVE_COMPONENT",
+        help="[Experimental] Component (`api-server` or `runner`) to serve, if not specified, will serve all components",
```
Comment (ssheng) [inline bentoml_cli/serve.py:25]: Should it be `api_server` with a underscore? `api_server` is how it appears in other references.

File: bentoml_cli/utils.py [hunk 5]
```diff
@@ -97,6 +97,7 @@ def wrapper(
             **kwargs: P.kwargs,
         ) -> t.Any:
             if config:
+                print("inject config:", config)
```
Comment (ssheng) [inline bentoml_cli/utils.py:100]:
```suggestion
```

File: requirements/docs-requirements.txt [hunk 6]
```diff
@@ -1,7 +1,7 @@
 # Docs dependencies
 -r tests-requirements.txt
-setuptools>=62.0.0
 sphinx==4.5.0
+setuptools==60.0.0
```
Comment (ssheng) [inline requirements/docs-requirements.txt:4]: What is the reason for the downgrade?

File: bentoml/_internal/bento/docker/entrypoint.sh (hunk 2 above)
Comment (bojiang) [reply bentoml/_internal/bento/docker/entrypoint.sh:14]: Not really

Comment (bojiang) [reply bentoml/_internal/bento/docker/entrypoint.sh:14]: `command`: 
`-z "$@"` -> true 
`[ "${1:0:1}" = '-' ]` -> false

File: requirements/docs-requirements.txt (hunk 6 above)
Comment (bojiang) [reply requirements/docs-requirements.txt:4]: ping @aarnphm

File: bentoml/_internal/bento/docker/entrypoint.sh (hunk 2 above)
Comment (sauyon) [reply bentoml/_internal/bento/docker/entrypoint.sh:14]: Minor style nit, but I think `[ $# -eq 1 ]` is clearer than `[ -z "$@" ]`.

File: requirements/docs-requirements.txt (hunk 6 above)
Comment (aarnphm) [reply requirements/docs-requirements.txt:4]: this has to do with the newer version of setuptools and it setup different kind of editables install. It is better to just lock setuptools for now.

**Comment by sauyon** [comment]
I think HTTP is more accurate than REST, since our APIs can pretty easily be used to make a non-RESTful server.

File: bentoml_cli/utils.py [hunk 7]
```diff
@@ -97,6 +97,7 @@ def wrapper(
...
                 load_global_config(config)
```
Comment (ssheng) [inline bentoml_cli/utils.py:102]:
```suggestion
```

**Review by ssheng — APPROVED** [review]

**Pull request merged**